### ISSUE TITLE
JSX Fragments are in-properly typed when not using `React.Fragment`

### ISSUE THREAD

**Issue by gkemp94** [microsoft/TypeScript#50429]
# Bug Report

### 🔎 Search Terms

React Fragment

### 🕗 Version & Regression Information
 
I believe this is a new inconsistency due to changes in React.

### ⏯ Playground Link

https://github.com/gkemp94/test-typescript-fragment

### Code
```
import React from "react";

const test = () => "asd";

// No Errors
const jsxWithJsxFragment = <>{test}</>;

// Type '() => string' is not assignable to type 'ReactNode'.
const jsxWithReactFragment = <React.Fragment>{test}</React.Fragment>;

```

### 🙁 Actual behavior

When using `<>` vs `<React.Fragment>` the children are not properly type checked

### 🙂 Expected behavior

When jsx: react is specified, to type check `<>` the same as `React.Fragment`

**Comment by MartinJohns** [comment]
[Playground link](https://www.typescriptlang.org/play?#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgIilQ3wG4AoctCAOwGd4YkG4BeOACgEo2A+AlHQAmZSgHoxcAHIQ4AUSg4odKrRYArOgA8A6sBgALAFLaAYlBQBzEEhrx2AHl4BvJgwC+DsbwrkJcABUATzAkOABybj44BihgGktwuGA6OBoIeEE6YEsaFAAjABswmFkYELDw5HQYGSEkcIA6VXp4TV19A2qMcysbOzY4B26YRt7rWxgXNxhPMRGxiwm7H3IgA)

**Comment by a-tarasyuk** [comment]
@RyanCavanaugh Should `JsxOpeningFragment` be part of `CallLikeExpression`? 

https://github.com/microsoft/TypeScript/blob/ed6889cd5b61b9fa5156018362d867def18e281d/src/compiler/types.ts#L2802-L2808

It does not have `tagName` _(which we use in `JsxOpeningLikeElement` to resolve call)_ and other common `CallLikeExpression` properties - `attributes`, `typeArguments`, etc. 

Or do we need to make a special kind of fragment factory (with a synthetic name) call resolving?

**Comment by DanielRosenwasser** [comment]
It sounds like the answer is "yes it should be a call-like expression", but also you will need to virtualize a target. I don't know if we still do that, but we have helpers like "getEffective____` for getting the arguments we should check against.

**Comment by merrywhether** [comment]
Could this be solved in a similar manner to #51328? Libraries/types could define `JSX.FragmentType` in a similar manner to `JSX.ElementType`.
